addon/components/bs-form.js

```diff
@@ -139,6 +139,19 @@ export default Ember.Component.extend({
   }),
 
   /**
+   * Validate hook that will either resolve if the model is valid or reject if it's not
+   *
+   * @param Function resolve
+   * @param Function reject
+   * @public
+   */
+  validate(resolve, reject) {
```

Comment:
I think this would be cleaner if it would just return a Promise by itself, instead of giving the promise callbacks as arguments!?